sdc/datatypes/hpat_pandas_dataframe_rolling_functions.py

```diff
@@ -78,45 +82,149 @@
+    other_length = 'len(get_dataframe_data(other, 0))' if other_columns else '0'
+    func_lines += [f'  length = max([{data_length}, {other_length}])']
+
+    for col in all_columns:
+        res_data = f'result_data_{col}'
+        if col in common_columns:
+            other_series = f'other_series_{col}'
+            method_kws['other'] = other_series
+            method_params = ', '.join(args + params2list(method_kws))
+            func_lines += [
+                f'  data_{col} = get_dataframe_data(self._data, {data_columns[col]})',
+                f'  other_data_{col} = get_dataframe_data(other, {other_columns[col]})',
+                f'  series_{col} = pandas.Series(data_{col})',
+                f'  {other_series} = pandas.Series(other_data_{col})',
+                f'  rolling_{col} = series_{col}.rolling({rolling_params})',
+                f'  result_{col} = rolling_{col}.corr({method_params})',
+                f'  {res_data} = result_{col}._data[:length]'
+            ]
+        else:
+            func_lines += [
+                f'  {res_data} = numpy.empty(length, dtype=float64)',
+                f'  {res_data}[:] = numpy.nan'
+            ]
+        results.append((col, res_data))
+
+    data = ', '.join(f'"{col}": {data}' for col, data in results)
+    func_lines += [f'  return pandas.DataFrame({{{data}}})']
+    func_text = '\n'.join(func_lines)
+
+    global_vars = {'numpy': numpy, 'pandas': pandas, 'float64': float64,
+                   'get_dataframe_data': get_dataframe_data}
+
+    return func_text, global_vars
+
+
+def df_rolling_method_tail_codegen(method_params, df_columns, method_name):
+    rolling_params = df_rolling_params_codegen()
+    method_params_as_str = ', '.join(method_params)
+
+    results = []
+    func_lines = []
+    for idx, col in enumerate(df_columns):
         res_data = f'result_data_{col}'
         func_lines += [
             f'  data_{col} = get_dataframe_data(self._data, {idx})',
             f'  series_{col} = pandas.Series(data_{col})',
-            f'  rolling_{col} = series_{col}.rolling({rolling_params_as_str})',
+            f'  rolling_{col} = series_{col}.rolling({rolling_params})',
             f'  result_{col} = rolling_{col}.{method_name}({method_params_as_str})',
             f'  {res_data} = result_{col}._data[:len(data_{col})]'
         ]
         results.append((col, res_data))
 
     data = ', '.join(f'"{col}": {data}' for col, data in results)
     func_lines += [f'  return pandas.DataFrame({{{data}}})']
+
+    return func_lines
+
+
+def df_rolling_method_codegen(method_name, self, with_other=False, args=None, kws=None):
+    args = args or []
+    kwargs = kws or {}
+
+    impl_params = ['self'] + args + params2list(kwargs)
+    impl_params_as_str = ', '.join(impl_params)
+
+    impl_name = f'_df_rolling_{method_name}_impl'
+    func_lines = [f'def {impl_name}({impl_params_as_str}):']
+
+    if 'other' in kwargs and not with_other:
```

Comment:
From what I see below 'other' will always be in kwargs, am I wrong? Can this part of the check be omitted?
Also I still think it's better to pass other as a type here (but not as a boolean variable that indicates that other has certain types) and make the decision here rather than on the call site, e.g.
```suggestion
    other_is_none_or_default = isinstance(other, (Omitted, NoneType)) or other is None
    if other_is_none_or_default:
```

Reply:
OK, it looks like 'other' wont be in kwargs for other rolling methods. So with my suggestions we reduce the code a bit, but actually make it more complex. Let's keep the original version then, but keep consistent naming of gen functions, e.g. df_rolling_method_other_df_codegen, df_rolling_method_other_none_codegen, etc.